libcudacxx/test/libcudacxx/cuda/functional/min_max_common.h

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _TEST_LIBCUDACXX_CUDA_FUNCTIONAL_MIN_MAX_COMMON_H
+#define _TEST_LIBCUDACXX_CUDA_FUNCTIONAL_MIN_MAX_COMMON_H
+
+#include <cuda_runtime.h>
+
+#include <cstdio>
+#include <cstdlib>
+
+#define CUDA_SAFE_CALL(...)                                                           \
```

Comment:
Our libcu++ tests inherently do run everything within main both on host and within a kernel.

So you should not need any of that. Please have a look at e.g libcudacxx/test/libcudacxx/std/utilities/function.objects/comparisons/greater.pass.cpp

The `__global__` kernel is just to ensure that we properly suppress execution space checks

Nowadays we put everything into a `__host__ __device__ TEST_CONSTEXPR_CXX14 bool test()` and then do

```cpp
int main(int, char**)
{
 test();
#if TEST_STD_VER >= 2014
  static_assert(test(), "");
#endif // TEST_STD_VER >= 2014
  return 0;;
}
```

Reply:
Fixed.